File: LinkedList/CompareLinkedList.py
#Compare two linked list

class Node:
    def __init__(self, data,next):
        self.data = data
        self.next = next

class LinkedList:
    def __init__(self):
        self.currNode = None
    
    def add(self,data):
        newNode = Node(data,self.currNode)
        self.currNode = newNode
        return self.currNode
    
    def printAll(self,head):
        while head:
            print head.data
            head = head.next
    
    def CompareLists(self,headA, headB):
        if headA is None or headB is None:
            return 0
        else:
            nodeA = headA
            nodeB = headB
            unmatched = 0 
            while nodeA is not None and nodeB is not None:
                if nodeA.data != nodeB.data:
                    unmatched = 2
                nodeA= nodeA.next
                nodeB=nodeB.next
            if unmatched == 2:
                return 0
            else:
                if nodeA is None and nodeB is not None:
                    return 0
                elif nodeB is None and nodeA is not None:
                    return 0
                else:
                    return 1    
    

if __name__ == "__main__":
    li = LinkedList()
    li.add(2)
    li.add(4)
    head = li.add(7)
    #li.printAll(head)

    li2 = LinkedList()
    #li2.add(1)
    li2.add(4)
    head2 = li2.add(7)
    #li2.printAll(head2)

    print li2.CompareLists(head,head2)
    

File: DS/LinkedList/CompareLinkedList.js
//Compare Linked List

function Node(data, next) {
    this.data = data;
    this.next = next;
}

function LinkedList() {
    this.currNode = null;
}

LinkedList.prototype = {
    add: function(data) {
        var newNode = new Node(data, this.currNode);
        this.currNode = newNode;
        return this.currNode;
    },
    print: function(head) {
        while (head) {
            console.log(head.data);
            head = head.next;
        }
    },
    compare: function(headA, headB) {
        if (headA == null || headB == null) {
            return 0;
        } else {
            var unmatched = 0;
            while (headA && headB) {
                if (headA.data != headB.data) {
                    unmatched = 1;
                }
                headA = headA.next;
                headB = headB.next;
            }

            if (unmatched == 1) {
                return 0;
            } else {
                if (headA == null && headB != null) {
                    return 0;
                } else if (headB == null && headA != null) {
                    return 0;
                } else {
                    return 1;
                }
            }
        }
    }
}

var li = new LinkedList();
li.add(2);
li.add(6);
var headA = li.add(3);
//li.print(headA);
var li2 = new LinkedList();
li2.add(2);
li2.add(6);
var headB = li2.add(3);
console.log(li2.compare(headA, headB));